### DISCUSSION TITLE
How to preserve sideset names with FancyExtruderGenerator

### DISCUSSION THREAD

**Discussion by aprilnovak** [idaholab/moose#16623] [Q&A Meshing] [answered]
Hi all,

I'd like to make a pincell mesh, starting from a circle mesh generated with the `AnnularMeshGenerator`, that is then extruded in the z-direction. I want to apply a boundary condition on the surface of the pincell, but am unsure how to access that sideset following the extrusion process.

`AnnularMeshGenerator` labels the circle surface as a sideset with name `rmax`. But once I extrude it in the z-direction and try to apply a BC on `rmax`, I get a: `the following side set ids do not exist on the mesh` error. I feel like I'm just missing something basic here, but none of the sideset generators jump out at me as being capable of creating a sideset for the curved cylinder surface (b/c the normal isn't fixed). All the tests for the extrude-type mesh generators also only apply BCs on the bottom/top sidesets that you can name in the generator itself.

Any suggestions for how to get a valid sideset on the cylinder surface? Here's my mesh:

```
[Mesh]
  [circle]
    type = AnnularMeshGenerator
    nr = 8
    nt = 8
    rmin = 0
    rmax = 0.4E-2
    growth_r = 1.2
  []

  [extrude]
    type = FancyExtruderGenerator
    input = circle
    heights = '0.8'
    num_layers = '10'
    direction = '0 0 1'
    bottom_sideset = '100'
    top_sideset = '101'
  []
[]

[Outputs]
  exodus = true
[]
```

Thanks!

-April

**Comment by GiudGiud** [comment, marked as answer]
Hi April,

@friedmud made such meshes for BEAVRS, you may see them in the mockingbird repository. I can't add you rn, maybe @smharper ?
@smharper may also be able to help directly as he used the FancyExtruder recently. 

Best,
Guillaume

**Reply by aprilnovak** [reply]
I figured out my problem! It _seems_ that the FancyExtruder doesn't retain sideset _names_ when you extrude, but does preserve sideset _numbers_. I was trying to apply BCs based on names, which I had assumed would transfer correctly. 

Is there a reason why the sideset names aren't also transferred? I think it could be very helpful to transfer those as well as the sideset IDs.

**Reply by aeslaughter** [reply]
Probably just an oversight.
